apps/framework-docs-v2/src/components/mdx/interactive/customize-panel.tsx

```
@@ -1,67 +1,173 @@
-    </Card>
+  const [showCustomizer, setShowCustomizer] = useState(false);
+  const [selections, setSelections] = useState<Record<string, string> | null>(
+    null,
   );
+  const [isClient, setIsClient] = useState(false);
+
+  // Stabilize fieldIds to avoid unnecessary re-runs when array reference changes
+  const fieldIdsKey = fieldIds?.join(",") || "";
+
+  // Check for existing selections on mount
+  useEffect(() => {
```

Biome (2.3.14)

[error] This hook does not specify its dependency on fieldIds.

This dependency is being used here, but is not specified in the hook dependency list.

This dependency is being used here, but is not specified in the hook dependency list.

React relies on hook dependencies to determine when to re-compute Effects.
Failing to specify dependencies can result in Effects not updating correctly when state changes.
These "stale closures" are a common source of surprising bugs.
Unsafe fix: Add the missing dependency to the list.


(lint/correctness/useExhaustiveDependencies)